src/test/java/com/tenx/ms/retail/rest/TestStoreController.java

```diff
@@ -0,0 +1,212 @@
+package com.tenx.ms.retail.rest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tenx.ms.commons.config.Profiles;
+import com.tenx.ms.commons.rest.RestConstants;
+import com.tenx.ms.commons.rest.dto.ResourceCreated;
+import com.tenx.ms.commons.tests.AbstractIntegrationTest;
+import com.tenx.ms.retail.RetailServiceApp;
+import com.tenx.ms.retail.store.rest.dto.Store;
+import org.apache.commons.io.FileUtils;
+import org.flywaydb.test.annotation.FlywayTest;
+import org.flywaydb.test.junit.FlywayTestExecutionListener;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.TestRestTemplate;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by goropeza on 25/08/16.
+ */
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebIntegrationTest(randomPort = true)
+@SpringApplicationConfiguration(classes = RetailServiceApp.class)
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class, FlywayTestExecutionListener.class})
+@ActiveProfiles(Profiles.TEST_NOAUTH)
+public class TestStoreController extends AbstractIntegrationTest {
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private static final String API_VERSION = RestConstants.VERSION_ONE;
+
+    public static final String REQUEST_URI = "%s" + API_VERSION + "/stores/";
+
+    private final RestTemplate template = new TestRestTemplate();
+
+    @Value("classpath:assets/new_store.json")
+    private File newStoreRequest;
+
+    @Value("classpath:assets/update_store.json")
+    private File updateStoreRequest;
+
+    @Value("classpath:assets/new_store_empty_name.json")
+    private File newStoreEmptyNameRequest;
+
+    @Value("classpath:assets/new_store_null_name.json")
```

Comment:
It's very minor but if you have separated test classes for different component, then it would be nice to also separate these request json files under their own folder.